lib/execution_segment.go

```diff
@@ -384,13 +385,82 @@ func (ess ExecutionSegmentSequence) String() string {
 	return strings.Join(result, ",")
 }
 
+func (ess ExecutionSegmentSequence) Len() int {
+	return len(ess)
+}
+
+func (ess ExecutionSegmentSequence) Less(i, j int) bool {
+	// Yes this Less is actually More, but we want it sorted in descending order and the alternative is to reverse it after sort
```

Comment:
I think this should be a comment of the method :wink: Or, actually, you probably should just use [`sort.SliceStable`](https://golang.org/pkg/sort/#SliceStable), instead of writing publicly wrong methods for something so specific.

Reply:
done in 9174195  ... I reused the variable name as I really don't need the original after that point